tests/test_init.py

```
@@ -181,245 +151,1362 @@ async def test_async_setup_entry_success(
+        remove_call.args[0] for remove_call in entity_registry.async_remove.call_args_list
+    }
+    assert stale_native_firewall.entity_id in removed_entity_ids
+    assert current_native_firewall.entity_id in removed_entity_ids
+    assert native_nat.entity_id in removed_entity_ids
+    ph_hass.config_entries.async_update_entry.assert_called_once_with(entry, version=5)
+
+
+@pytest.mark.asyncio
+async def test_migrate_4_to_5_non_granular_entry_defaults_sync_filters_and_nat_enabled(
+    monkeypatch: pytest.MonkeyPatch,
+    ph_hass: Any,
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `ph_hass`

(ANN401)